plugin/storage/badger/factory_test.go

```diff
@@ -166,28 +167,28 @@ func TestBadgerMetrics(t *testing.T) {
 	mFactory := metricstest.NewFactory(0)
 	f.Initialize(mFactory, zap.NewNop())
 	assert.NotNil(t, f.metrics.badgerMetrics)
-	_, found := f.metrics.badgerMetrics["badger_v3_memtable_gets_total"]
+	_, found := f.metrics.badgerMetrics["badger_get_num_memtable"]
 	assert.True(t, found)
 
 	waiter := func(previousValue int64) int64 {
 		sleeps := 0
 		_, gs := mFactory.Snapshot()
-		for gs["badger_v3_memtable_gets_total"] == previousValue && sleeps < 8 {
+		for gs["badger_get_num_memtable"] == previousValue && sleeps < 8 {
```

Comment:
What's the source of these changes? I understand expvar that may be defined by badger lib directly? But this is coming through our metrics api.

Reply:
the expvar a stored in you go pkg folder so inside that it is specified at `pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.5/y/metrics.go` and for v4 i looked at `pkg/mod/github.com/dgraph-io/badger/v4@v4.2.0/y/metrics.go` for equivalent variable names

Reply:
Ah, I see what happens - we have `factory.metricsCopier` function that re-publishes Badger's expvar metrics into our metrics factory